quantum/quantum_task_queue.h

```diff
@@ -85,26 +86,76 @@ class TaskQueue : public IQueue
     bool isIdle() const final;
 
 private:
-    TaskListIter advance();
+    struct WorkItem
+    {
+        WorkItem(TaskPtr task,
+                 TaskListIter iter,
+                 bool isBlocked,
+                 unsigned int blockedQueueRound);
+        
+        TaskPtr _task;                   // task pointer
```

Comment:
I would add default initializers for all members

Reply:
The _iter member is an iterator, which by default should be TaskQueue::_runQueue.end(), so that won't work unless I make it a pointer.